public/app/features/dashboard/state/DashboardModel.ts

```diff
@@ -167,7 +174,16 @@ export class DashboardModel {
 
     this.addBuiltInAnnotationQuery();
     this.sortPanelsByGridPos();
-    this.hasChangesThatAffectsAllPanels = false;
+    this.panelsAffectedByVariableChange = null;
+    this.appEventsSubscription = new Subscription();
+    this.lastRefresh = Date.now();
+    this.appEventsSubscription.add(appEvents.subscribe(VariablesChanged, this.variablesChangedHandler.bind(this)));
+    this.appEventsSubscription.add(
+      appEvents.subscribe(VariablesTimeRangeChanged, this.variablesTimeRangeChangedHandler.bind(this))
```

Comment:
the naming/meaning of this event is a bit confusing, but not sure what would be more clear can only think of "VariablesFinishedProcesssingTimeRangeChange"

Reply:
Good suggestion, I like that.